pkg/process/procutil/process_linux.go

```diff
@@ -104,24 +114,39 @@ func (p *Probe) StatsForPIDs(pids []int32, now time.Time) (map[int32]*Stats, err
 		statInfo := p.parseStat(pathForPID, pid, now)
 		memInfoEx := p.parseStatm(pathForPID)
 
-		statsByPID[pid] = &Stats{
-			CreateTime:  statInfo.createTime,      // /proc/[pid]/stat
-			Status:      statusInfo.status,        // /proc/[pid]/status
-			Nice:        statInfo.nice,            // /proc/[pid]/stat
-			OpenFdCount: p.getFDCount(pathForPID), // /proc/[pid]/fd, requires permission checks
-			CPUTime:     statInfo.cpuStat,         // /proc/[pid]/stat
-			MemInfo:     statusInfo.memInfo,       // /proc/[pid]/status
-			MemInfoEx:   memInfoEx,                // /proc/[pid]/statm
-			CtxSwitches: statusInfo.ctxSwitches,   // /proc/[pid]/status
-			NumThreads:  statusInfo.numThreads,    // /proc/[pid]/status
-			IOStat:      ioInfo,                   // /proc/[pid]/io, requires permission checks
+		stats := &Stats{
+			CreateTime:  statInfo.createTime,    // /proc/[pid]/stat
+			Status:      statusInfo.status,      // /proc/[pid]/status
+			Nice:        statInfo.nice,          // /proc/[pid]/stat
+			CPUTime:     statInfo.cpuStat,       // /proc/[pid]/stat
+			MemInfo:     statusInfo.memInfo,     // /proc/[pid]/status
+			MemInfoEx:   memInfoEx,              // /proc/[pid]/statm
+			CtxSwitches: statusInfo.ctxSwitches, // /proc/[pid]/status
+			NumThreads:  statusInfo.numThreads,  // /proc/[pid]/status
+		}
+		if withPerm {
+			stats.OpenFdCount = p.getFDCount(pathForPID) // /proc/[pid]/fd, requires permission checks
+			stats.IOStat = ioInfo                        // /proc/[pid]/io, requires permission checks
```

Comment:
Should `p.parseIO()` also be moved into this block?

Reply:
Good catch I copy pasted and messed up the order.